tests/e2e/lib/virt_helpers.go

```diff
@@ -0,0 +1,408 @@
+	group := operatorsv1.OperatorGroup{}
+	err := v.Client.Get(context.TODO(), client.ObjectKey{Namespace: v.Namespace, Name: "kubevirt-hyperconverged-group"}, &group)
+	if err != nil {
+		return false
+	}
+	return true
+}
+
+// Checks if there is a virtualization subscription
+func (v *VirtOperator) checkSubscription() bool {
+	subscription := operatorsv1alpha1.Subscription{}
+	err := v.Client.Get(context.TODO(), client.ObjectKey{Namespace: v.Namespace, Name: "hco-operatorhub"}, &subscription)
+	if err != nil {
+		return false
+	}
+	return true
+}
+
+// Checks if the ClusterServiceVersion status has changed to ready
+func (v *VirtOperator) checkCsv() bool {
+	subscription, err := v.GetOperatorSubscription()
+	if err != nil {
+		if err.Error() == "no subscription found" {
+			return false
+		}
+	}
+
+	return subscription.CsvIsReady(v.Client)
+}
+
+// CheckHco looks for a HyperConvergedOperator and returns whether or not its
+// health status field is "healthy". Uses dynamic client to avoid uprooting lots
+// of package dependencies, which should probably be fixed later.
+func (v *VirtOperator) checkHco() bool {
+	resourceId := schema.GroupVersionResource{
+		Group:    "hco.kubevirt.io",
+		Resource: "hyperconvergeds",
+		Version:  "v1beta1",
+	}
+
+	unstructuredHco, err := v.Dynamic.Resource(resourceId).Namespace(v.Namespace).Get(context.Background(), "kubevirt-hyperconverged", v1.GetOptions{})
+	if err != nil {
+		log.Printf("Error getting HCO: %v", err)
+		return false
+	}
+
+	health, ok, err := unstructured.NestedString(unstructuredHco.UnstructuredContent(), "status", "systemHealthStatus")
+	if err != nil {
+		log.Printf("Error getting HCO health: %v", err)
+		return false
+	}
+	if !ok {
+		log.Printf("HCO health field not populated yet")
+		return false
+	}
+	log.Printf("HCO health status is: %s", health)
+
+	return health == "healthy"
+}
+
+// Creates the target virtualization namespace, likely openshift-cnv or kubevirt-hyperconverged
+func (v *VirtOperator) installNamespace() error {
+	err := v.Client.Create(context.Background(), &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: v.Namespace}})
+	if err != nil {
+		log.Printf("Failed to create namespace %s: %v", v.Namespace, err)
+		return err
+	}
+	return nil
+}
+
+// Creates the virtualization operator group
+func (v *VirtOperator) installOperatorGroup() error {
+	group := &operatorsv1.OperatorGroup{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "kubevirt-hyperconverged-group",
+			Namespace: v.Namespace,
+		},
+		Spec: operatorsv1.OperatorGroupSpec{
+			TargetNamespaces: []string{
+				v.Namespace,
+			},
+		},
+	}
+	err := v.Client.Create(context.Background(), group)
+	if err != nil {
+		if !strings.Contains(err.Error(), "already exists") {
+			log.Printf("Failed to create operator group: %v", err)
+			return err
+		}
+	}
+	return nil
+}
+
+// Creates the subscription, which triggers creation of the ClusterServiceVersion.
+func (v *VirtOperator) installSubscription() error {
```

Comment:
Similar to above comment.

Our CatalogSource, CatalogSourceNamespace could be another type, this way we can have easy way of installing operators from there? Just a thought, which may not be worth changing.

Reply:
This one I am less sure about, because it would likely look pretty much the same. It's pretty much just passing eight arguments in a Subscription to Create, and checking the result. It would be nice to have a whole factored-out Operator installation module, but I don't know that it's worth building that until we have more operators to install.